### DISCUSSION TITLE
Change new line behavior for one custom format for Tinymce in Directus

### DISCUSSION THREAD

**Discussion by brendengerber** [tinymce/tinymce#9551] [General]
I'm using Tinymce with Directus and trying to add a custom formatting option for a blog that I will use to format blocks of code. Currently I have it set up like this and it's working almost as intended (entered into the Custom Formats section in Directus)

```
[
    {
        "title": "Code",
        "inline": "span",
        "classes": "code_snippet"
    }
]
```

The issue that I am having is that when I hit enter it creates a new span. The issue this causes is that now each line gets a border around it, when the intended behavior is to have the border go around the entire block. Is there a way to force Tinymce to add a br tag instead of creating a new span for ONLY this code block (I want to keep the default behavior for other things like p and h1)?

As a follow up, is there away to prevent the span from being nested inside a p tag (Ideally I would set the inline element of the custom format to p, but it doesn't work, so I have used span instead)?

**Comment by mattcoz** [comment]
Try the configuration option "br_newline_selector". It's not documented, but it exists. It's a comma separated list and the default value is ".mce-toc h2,figcaption,caption". So, you could change it to ".mce-toc h2,figcaption,caption,span.code_snippet".
